src/main/java/redis/clients/util/Pool.java

```diff
@@ -48,6 +49,11 @@ public T getResource() {
     try {
       return internalPool.borrowObject();
     } catch (NoSuchElementException nse) {
+      if (null == nse.getCause()) { // The exception was caused by an exhausted pool
```

Comment:
This same code snippet could be validated by the program catching the exception. I don't really see a benefit by throwing this exception. Another thing is:, are you sure that every time getCasue returns null is because the pool is exhausted?. How do you know this?